packages/plugin-team/src/team.js

```diff
@@ -0,0 +1,286 @@
+
+  },
+
+  /**
+   * Archive a team or team conversation.
+   * @param {TeamObject|ConversationObject} target team or team conversation that should be archived
+   * @param {Conversation~ActivityObject} activity
+   * @returns {Promise} Resolves with the posted activity
+   */
+  archive(target, activity) {
+
+  },
+
+  /**
+   * Create a team
+   * @param {NewTeamObject} params
+   * @returns {Promise} Resolves with the created team
+   */
+  create(params) {
+
+    if (!params.displayName) {
+      return Promise.reject(new Error(`'params.displayName' is required`));
+    }
+
+    if (!params.participants || params.participants.length === 0) {
+      return Promise.reject(new Error(`'params.participants' is required`));
+    }
+
+    return Promise.all(params.participants.map((participant) => this.spark.user.asUUID(participant, {create: true})))
+      .then((participants) => {
+        participants.unshift(this.spark.device.userId);
+        params.participants = uniq(participants);
+      })
+      .then(() => this._prepareTeam(params))
+      .then((payload) => this.request({
+        method: `POST`,
+        api: `conversation`,
+        resource: `teams`,
+        body: payload
+      }))
+      .then((res) => res.body);
+  },
+
+  /**
+   * Create a conversation within a team. Currently does not support
+   * activities besides add (ie no post or share activities).
+   * @param {TeamObject} team
+   * @param {NewConversationObject} params
+   * @param {Object} options
+   * @param {boolean} options.includeAllTeamMembers
+   * @returns {Promise} Resolves with the newly created conversation
+   */
+  createConversation(team, params, options) {
+    options = options || {};
+
+    if (!params.displayName) {
+      return Promise.reject(new Error(`\`params.displayName\` is required`));
+    }
+
+    return this._inferTeamUrl(team)
+      .then(() => this._ensureGeneralConversation(team))
+      .then((generalConversation) => Promise.all(
+        params.participants.map((participant) => this.spark.user.asUUID(participant, {create: true}))
+      )
+        .then((participants) => {
+          participants.unshift(this.spark.device.userId);
+          params.participants = uniq(participants);
+        })
+        .then(() => this._prepareTeamConversation(generalConversation, params))
+      )
+      .then((payload) => this.spark.request({
+        method: `POST`,
+        uri: `${team.url}/conversations`,
+        qs: pick(options, `includeAllTeamMembers`),
+        body: payload
+      }))
+      .then((res) => res.body);
+  },
+
+  /**
+   * Retrieve a single team
+   * @param {Object|Team~TeamObject} team
+   * @param {string} team.id
+   * @param {Object} options
+   * @param {boolean} options.includeTeamConversations
+   * @param {boolean} options.includeTeamMembers
+   * @returns {Promise} Resolves with the requested team
+   */
+  get(team, options) {
+    options = options || {};
+
+    return this._inferTeamUrl(team)
```

Comment:
I see. We used to whitelist params and now we're accepting all params. Seems like a good choice - now we won't need a change whenever a new param is added to the service. Any reason in particular we essentially blacklist these ones here? https://github.com/ciscospark/spark-js-sdk/blob/master/packages/plugin-conversation/src/conversation.js#L206